### DISCUSSION TITLE
[.NET9 + WASM] LinkerConfig.xml no longer respected?!?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#18919] [Q&A]
### Current behavior

I isolated a repo from the issues I'm chasing while upgrading my UNO solution from .NET8 -> .NET9 [here](https://github.com/unoplatform/uno/discussions/18778). After upgrading I'm getting tons of linker warnings on my code although all my assemblies are (still) excluded in `LinkerConfig.xml`.

Please see repo below:
- there is a single code line which causes a warning
- however, the custom assembly is excluded in `LinkerConfig.xml`
[Minimal.9.zip](https://github.com/user-attachments/files/17878879/Minimal.9.zip)

Why am I getting the warning? Is the linker no longer respecting LinkerConfig.xml?

**Comment by jeromelaban** [comment]
Thanks for the report.

These new .NET 9 warnings are coming from the IL Linker, which analyzes the code regardless of its exclusion status. Whether it's a bug or not is debatable, yet in all cases is a behavior that comes from the official .NET SDK/Runtime, and we do not have control over this.

We're still going to be fixing those issues for an upcoming release of uno, for those that are coming from uno itself.

**Reply by DierkDroth** [reply]
@jeromelaban thanks for your feedback. So, we actually don't know and can't be sure that the exclusions work or not, correct?

**Reply by DierkDroth** [reply]
Additionally, what would be the correct place to raise this .NET issue? You know?

**Reply by jeromelaban** [reply]
> @jeromelaban thanks for your feedback. So, we actually don't know and can't be sure that the exclusions work or not, correct?

The exclusions are working as expected, code is removed as configured, but the code analyzer does not take exclusions into account to report warnings. You can ask in dotnet/runtime, they will redirect to the right place.

**Comment by DierkDroth** [comment]
FYI for anyone interested: I started a conversation [here](https://github.com/dotnet/runtime/issues/110181)

Closing this issue
